function-maven-plugin/src/main/java/com/microsoft/azure/maven/function/AbstractFunctionMojo.java

```diff
@@ -40,11 +52,19 @@ public String getRegion() {
         return region;
     }
 
-    public String getFunctionName() {
-        return functionName;
+    public String getDeploymentStageDirectory() {
+        return Paths.get(getBuildDirectoryAbsolutePath(),
+                "azure-functions",
+                getAppName()).toString();
     }
 
-    public List<Resource> getResources() {
-        return resources;
+    public FunctionApp getFunctionApp() {
+        FunctionApp app = null;
+        try {
+            app = getAzureClient().appServices().functionApps().getByResourceGroup(getResourceGroup(), getAppName());
+        } catch (Exception e) {
+            // Swallow exception
```

Comment:
@ZhijunZhao  This is to check the existence of the Function App. Azure Java SDK doesn't provide an API for such check.